src/json_stream/tokenizer/strings.py

```diff
@@ -0,0 +1,169 @@
+import io
+import unicodedata
+from typing import Union
+from io import DEFAULT_BUFFER_SIZE
+
+STRING_ESCAPE_CODES = {
+    '\\': '\\',
+    '/': '/',
+    '"': '"',
+    'b': '\b',
+    'f': '\f',
+    'n': '\n',
+    't': '\t',
+    'r': '\r'
+}
+
+SURROGATE = 'Cs'
+
+CHAR = 1
+STRING_ESCAPE = 2
+UNICODE = 4
+UNICODE_SURROGATE_START = 5
+UNICODE_SURROGATE_STRING_ESCAPE = 6
+UNICODE_SURROGATE = 7
+
+
+class JsonStringReader(io.TextIOBase):
+    def __init__(self, stream: io.TextIOBase, initial_buffer=''):
+        self.stream = stream
+        self.buffer = initial_buffer
+        self.readline_buffer = ''
+        self.unicode_buffer = ''
+        self.state = CHAR
+        self.end_of_string = False
+        self.index = 0
+
+    @property
+    def complete(self):
+        return self.end_of_string and not self.readline_buffer
+
+    def readable(self) -> bool:
+        return True
+
+    def read(self, size: Union[int, None] = None) -> str:
+        result = ''
+        length = DEFAULT_BUFFER_SIZE
+        while not self.complete and (size is None or not result):
+            if size:
+                length = size - len(result)
+            result += self._read_chunk(length)
+        return result
+
+    def _read_chunk(self, size: Union[int, None] = ...) -> str:
```

Comment:
What does the ellipsis `...` as `size`'s default value here signify? Shouldn't it just be a mandatory parameter?

Reply:
I think this comes from when PyCharm created the method signature automatically. It bases type information like that on the typeshed project:
https://github.com/python/typeshed/blob/3d352e85e5e107c8a7e0f59854b23cebc8940f5e/stdlib/io.pyi#L145 me 

Now, why the typeshed type signature for `TextIOBase.read()` has that elipsis is another question...

Reply:
I've pushed a change, `_read_chunk()` is always called with an int, so I've made that explicit.